### PR TITLE
feat: create new assert function

### LINKED ISSUE #675: Assert that a variable is not null
**Is your feature request related to a problem? Please describe.**
The engine's behavior for non-existing variables will change with https://github.com/camunda/feel-scala/issues/582 and https://github.com/camunda/feel-scala/issues/674. Instead of failing the evaluation, a non-existing variable will be replaced with `null`. 

So, the evaluation will be `null`-friendly. If the engine can't evaluate the expression successfully, it will return `null`. 

This behavior is aligned with the DMN spec. However, there could be cases that require a verification that a variable exists or is not `null`. 

A special case is the migration of existing expression to the new behavior. For example, if a process uses an expression and assumes that the expression fails if the variable doesn't exist (e.g. on an output variable mapping).

**Describe the solution you'd like**
Add a new built-in function to verify that a variable or value is not `null`.

```
assert not null(value: x)
// if x is null, the evaluation fails
// if x is not null, the function returns x

assert not null(value: x, cause: "x is null")
// with an optional failure message
``` 

Or, a more general built-in function to verify that a condition is met.

```
assert(value: x, condition: x != null)
// if the condition is false or null, the evaluation fails
// if the condition is true, the function returns the value 

assert(value: x, condition: x != null, cause: "x is null")
// with an optional failure message
```

**Related issues**

* Camunda Autormation Platform 7: 
* Zeebe broker:

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

Create new `assert(value, condition)` function in order to allow users to verify if a variable met a certain condition, and return an error if the condition is not fulfilled. This function should help users to migrate to 8.3 easily if they rely on validation error.

Users can provide also a custom error message by `assert(value, condition, cause)`

## Related issues

closes #675

### CODE REVIEW

**Review by saig0 — CHANGES_REQUESTED** [review]
@nicpuppa well done. :tada: 

I found a few minor issues and one major issue. The function doesn't work in all cases. Please have a look at my comments. :cookie:

**Comment by saig0** [comment]
@nicpuppa it seems that the SLA bot doesn't like your commits. Please check if the email address of the commits match your GitHub account.

File: src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala [hunk 1]
```diff
@@ -205,6 +205,9 @@ class FeelInterpreter {
       case FunctionInvocation(name, params) =>
         withFunction(findFunction(context, name, params),
           f => invokeFunction(f, params) match {
+            case ValError(failure) if name == "assert" =>
+              error(EvaluationFailureType.FUNCTION_INVOCATION_FAILURE, s"Failed to invoke function '$name': $failure")
```
[the remark this answers is not in the document]
Comment (nicpuppa) [reply src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala]: Thank you for the suggestion, you are right 💪

File: src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala [hunk 2]
```diff
@@ -205,6 +205,9 @@ class FeelInterpreter {
...
+              ValError(failure)
```
[the remark this answers is not in the document]
Comment (nicpuppa) [reply src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala:210]: This was a tricky one 👀 🤯

**Comment by nicpuppa** [comment]
Hi @saig0 thanks for the review 🚀 

I applied all the suggestions, please have another look 👀  Sorry for the CLA, new pc 😂

File: src/main/scala/org/camunda/feel/FeelEngine.scala [hunk 3]
```diff
@@ -161,6 +161,11 @@ class FeelEngine(
           failure = Failure(s"failed to evaluate expression '${exp.text}': $cause"),
           suppressedFailures = context.failureCollector.failures
         )
+      case _ if containsAssertionError(context) =>
```
Comment (saig0) [inline src/main/scala/org/camunda/feel/FeelEngine.scala:164]: :x: We should move this case to the top to ensure that we produce always the same failure message.

Currently, the failure message is different if the `assert()` function is wrapped inside another function or not.

**Review by saig0 — CHANGES_REQUESTED** [review]
@nicpuppa nice! :+1: 

One remaining task is to provide the custom failure message to the user. Please have a look at my comments. :cake:

**Comment by nicpuppa** [comment]
Applied all suggestions, please have another look 👀

**Review by saig0 — APPROVED** [review]
@nicpuppa looks good. :+1: 

Before merging, please fix the test failure. I guess that you rebased on `main` but didn't take all the changes. On `main`, this test case is ignored.

**Pull request merged**